### DISCUSSION TITLE
Do you have a SkiaShell simple example?

### DISCUSSION THREAD

**Discussion by lukewire129** [taublast/DrawnUi#4] [Q&A] [answered]
I'm trying to build something based on 'SkiaShell', but I'm not sure how to do it.
is there a simple example?

**Comment by lukewire129** [comment, marked as answer]
https://github.com/taublast/AppoMobi.Maui.DrawnUi.Demo/blob/main/src/AllDrawn/AppShell.xaml

**Comment by taublast** [comment]
Would definitely need to add more demos for shell, as its docs are non complete at all.

Some points of interest:

SkiaShell can be subclassed to override some virtuals, more customizations will come with time for sure. It has now some quick static properties to be modified at launch time for tuning, for example:
```
 public static Color ToastBackgroundColor = Color.Parse("#CC000000");
 public static Color ToastTextColor = Colors.White;
 public static string ToastTextFont = null;
 public static int ToastTextFontWeight = 0;
 public static double ToastTextSize = 16.0;
 public static double ToastTextMargins = 24.0;
 public static float PopupsAnimationSpeed = 150;
 public static bool LogEnabled = false;
```

When you push pages with `GoToAsync` you can define inside the page Xaml the presentation mode, for example:
`Shell.PresentationMode="ModalAnimated"`
then the route will be open as a modal instead of as a page.

At the same time you can push a modal from the route, for example:
```
  await Shell.PushModalAsync(AppRoutes.PaywallStart.Route, false, false, null, new Dictionary<string, object>
  {
      {"navigatedFrom", Analytics.MainScreen.ToString()} 
  });
```

You can pass objects and parameters like in MAUI shell and catch them inside pages or viewModels either by using the attribute `QueryProperty` or by implementing `IQueryAttributable` interface, all like in the usual MAUI. The second case is more suitable for cases when you pass objects as parameters that cannot be encoded in the route URL, while the first case handles well string parameters like `"details?id=21"`.

Example A:

```
[QueryProperty("Id", "id")]
public class ItemDetailsViewModel : ProjectViewModel, IItemDetailsViewModel
{
    private string _Id;
    public string Id
    {
        get { return _Id; }
        set
        {
            var newValue = Uri.UnescapeDataString(value ?? string.Empty);
            if (_Id != newValue)
            {
                _Id = newValue;
                OnPropertyChanged();
                OnParametersSet(_Id);
            }
        }
    }
```

Example B:

```
public class EditCollectionItemViewModel : ProjectViewModel, IQueryAttributable
{

    public void ApplyQueryAttributes(IDictionary<string, object> query)
    {
        if (query != null)
        {
            query.TryGetValue("navigatedFrom", out var source);
            if (source != null)
            {
                NavigatedFrom = source.ToString();

                var parameters = new Dictionary<string, string>()
                {
                    { "from",  NavigatedFrom},
                };
                App.Singletons.Analytics.Log(AnalyticEvents.edit_indetification_shown,
                    parameters);
            }

            query.TryGetValue("item", out var item);
            if (item != null)
            {
                Item = item as CollectionItem;
            }
        }
    }
```
